internal/dataplane/parser/translate_utils.go

```diff
@@ -208,3 +201,16 @@ func (p *Parser) generateKongServiceFromBackendRef(
 
 	return service, nil
 }
+
+// maybePrependRegexPrefix takes a path string and returns it withthe regex prefix prepended if the Kong version
+// requires it and it is not already present. If those conditions are not met, it returns the original path string.
+func maybePrependRegexPrefix(path string) string {
+	if util.GetKongVersion().GTE(MinExplicitPathRegexKongVersion) {
+		if LegacyRegexPathExpression.FindString(path) != "" {
```

Comment:
IMO it is better to do the judging and prepend on the place where we generate the route. Because in the place we generate the route, we KNOW whether we are going to make a regex route. If we do the judge after the route is generated and became a string, there would be many unpredictable corner cases where we could make mistake.

Reply:
Can you clarify? We only have affirmative information about whether something is a regex or not for HTTPRoute. Ingress is entirely ambiguous as to whether a user-provided path is a regex or not. We don't really have the concept of regex routes on the Kong side either, you can mix regex and prefix paths within a single route.